File: Data Structures & Algorithms/valid-binary-search-tree/submission-9.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    List<Integer> arr;

    private void compute(TreeNode node) {
        if (node == null) {
            return;
        }
        compute(node.left);
        arr.add(node.val);
        compute(node.right);
    }
    public boolean isValidBST(TreeNode root) {
        arr = new ArrayList<>();
        compute(root); // in order traversal
        int prev = -1001;
        for (int val : arr) {
            if (val <= prev) {
                return false;
            }
            prev = val;
        }
        return true;

    }
}

File: Data Structures & Algorithms/binary-tree-right-side-view/submission-9.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def rightSideView(self, root: Optional[TreeNode]) -> List[int]:
        
        res = []
        def bfs(queue):
            if not queue:
                return
            res.append(queue[-1].val)
            traverse = []
            for node in queue:
                if node.left:
                    traverse.append(node.left)
                if node.right:
                    traverse.append(node.right)
            bfs(traverse)
        
        if not root:
            return res
        
        bfs([root])
        return res

            